internal/jobcontainers/storage.go

```diff
@@ -0,0 +1,94 @@
+package jobcontainers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/layers"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+// Trailing backslash required for SetVolumeMountPoint and DeleteVolumeMountPoint
+const sandboxMountFormat = `C:\C\%s\`
+
+func mountLayers(ctx context.Context, s *specs.Spec) error {
+	if s == nil || s.Windows == nil || s.Windows.LayerFolders == nil {
+		return errors.New("field 'Spec.Windows.Layerfolders' is not populated")
+	}
+
+	scratchFolder := s.Windows.LayerFolders[len(s.Windows.LayerFolders)-1]
+
+	if _, err := os.Stat(scratchFolder); os.IsNotExist(err) {
+		if err := os.MkdirAll(scratchFolder, 0777); err != nil {
+			return fmt.Errorf("failed to auto-create container scratch folder %s: %s", scratchFolder, err)
+		}
+	}
+
+	// Create sandbox.vhdx if it doesn't exist in the scratch folder.
+	if _, err := os.Stat(filepath.Join(scratchFolder, "sandbox.vhdx")); os.IsNotExist(err) {
+		if err := wclayer.CreateScratchLayer(ctx, scratchFolder, s.Windows.LayerFolders[:len(s.Windows.LayerFolders)-1]); err != nil {
+			return fmt.Errorf("failed to CreateSandboxLayer %s", err)
+		}
+	}
+
+	if s.Root == nil {
+		s.Root = &specs.Root{}
+	}
+
+	if s.Root.Path == "" {
```

Comment:
Can you explain that the use case is here? I assume it's so that you don't *have* to use traditional container images as part of job containers, but what happens in that case? Is there anything that needs to be setup to support that?

Reply:
This is pulled straight from `/internal/hcsoci.allocateWindowsResources`. There's nothing special about the custom images we produced that would affect this afaik

Reply:
I think I may have been referring to what happens in the case that `s.Root.Path` != ""?

Reply:
I am probably missing some context here but I have the same question as Kathryn's last comment. Can you explain @dcantah ?

Reply:
Root.Path should always be empty afaik. This is just pulled straight from the regular hcs containers code as nothing really changes spec wise for these.